logreader.go

```diff
@@ -70,16 +85,33 @@ func (r *LogReader) read(l []byte) (int, error) {
 		return written, err
 	}
 
+	if written > 0 {
+		// Check for an STX (Start of Text) ASCII control marker.
+		if !r.startOfText && bytes.Contains(l, []byte("\x02")) {
+			r.startOfText = true
+		}
+
+		// If we found an STX ASCII control character, start looking for
+		// the ETX (End of Text) control character.
+		if r.startOfText && bytes.Contains(l, []byte("\x03")) {
+			r.endOfText = true
+		}
+	}
+
 	// Check if we need to continue the loop and wait 500 miliseconds
 	// before checking if there is a new chunk available or that the
 	// run is finished and we are done reading all chunks.
 	if written == 0 {
-		done, err := r.done()
-		if err != nil {
-			return 0, err
-		}
-		if done {
-			return 0, io.EOF
+		if (r.startOfText && r.endOfText) || // The logstream finished without issues.
+			(r.startOfText && r.reads%10 == 0) || // The logstream terminated unexpectedly.
+			(!r.startOfText && r.reads%2 == 0) { // The logstream doesn't support STX/ETX.
```

Comment:
Should this just be `!r.startOfText && r.reads > 1` ?

Reply:
Yeah we can change that, since we now already have the exponential backoff. But it does mean generating twice as much API calls for setups that do not yet run the updated TBW. So not sure if we should do that.

For the UX it will be a little bit better, but the only moment when you will really notice this is when the logs are already done and you are waiting for the binary to exit. If we set this to `r.reads > 1` it will take max 2s, of we set it to `r.reads%2 == 0` it will take max 4s.

Hmm... 4s is pretty long, so let's make this max 2s. then.